### DISCUSSION TITLE
Filling small holes in an image efficiently

### DISCUSSION THREAD

**Discussion by ghost** [SixLabors/ImageSharp#2251] [Q&A] [answered]
Hi there! Firstly, thanks for the library. I'm really enjoying using it!

Is there a built-in way to fill in areas of an image? I see that there's methods that allow you to fill in paths or polygons which sounds good to me, but I don't see any efficient way of generating these objects without simply getting the boundary pixels of every one manually which would mean looping through the image and manually checking each pixel's neighbours... There must be a better way.

I've attached an image to show the kind of areas that I'm talking about. Note that there are black holes in white areas of the image and white holes in black areas.

![TestSecondClass](https://user-images.githubusercontent.com/50894004/193522633-bdf00be6-c092-401e-a829-9dfc2a50adeb.jpg)

**Comment by antonfirsov** [comment, marked as answer]
Generating polygons based on pixel graphic data is definitely beyond the scope of the ImageSharp and ImageSharp.Drawing libraries.

I would do the filling some other way anyways, maybe by flooding the pixels from a starting point, if you mean some mspaint-like filling here.

**Comment by brianpopow** [comment]
Just wanted to throw in that `MedianBlur` is pretty good at getting rid of the small errors:

```
image.Mutate(img => img.MedianBlur(3, false));
```

Example result:
![median](https://user-images.githubusercontent.com/38701097/193816155-e21263db-5f23-415a-bb66-0332d110c919.png)

It wont help with the larger wholes, though.
